control/pure_pursuit/src/pure_pursuit/pure_pursuit_lateral_controller.cpp

```diff
@@ -323,25 +406,39 @@ boost::optional<PpOutput> PurePursuitLateralController::calcTargetCurvature(
   }
 
   // Calculate target point for velocity/acceleration
-  const auto target_point = calcTargetPoint(pose);
-  if (!target_point) {
+
+  const auto closest_idx_result =
+    motion_utils::findNearestIndex(*output_tp_array_, pose, 3.0, M_PI_4);
+  if (!closest_idx_result) {
+    RCLCPP_ERROR(node_->get_logger(), "cannot find closest waypoint");
     return {};
   }
 
-  const double target_vel = target_point->longitudinal_velocity_mps;
+  const double && target_vel =
+    trajectory_resampled_->points.at(*closest_idx_result).longitudinal_velocity_mps;
```

Comment:
@brkay54 Any reason why you use Rvalue (`&&`) for this code? In my understanding, `&&` for a primitive value doesn't do anything in c++.

Reply:
You are right, just misunderstanding, I am gonna remove it.